### DISCUSSION TITLE
Lesson 3: Question on Tuples and Dot Notation

### DISCUSSION THREAD

**Discussion by widmung89** [smartcontractkit/full-blockchain-solidity-course-py#549] [Q&A]
![Tuple](https://user-images.githubusercontent.com/95630418/145739810-63e96e1f-1d2d-422d-b0bc-e30130f642e4.png)

Hi sorry, mega noob here, I've never done coding in my life. When using the "latestRoundData" function, I don't understand why we need to enter the entire 5 variables inside the tuple i.e. 

    ( uint80 roundId,
      int256 answer,
      uint256 startedAt,
      uint256 updatedAt,
      uint80 answeredInRound
    ) = priceFeed.latestRoundData();
      return uint256(answer);

When you use "priceFeed.latestRoundData();", isn't it already understood that "latestRoundData" contains all 5 tuple elements? Why can't I write something like "return priceFeed.latestRoundData([uint256 answer]);" or something along that line?

---------------------------------------------------------------------------------------------------------------------------------------------------

Also, I am confused when the Dot Notation should be used. I see that in the case of Javascript, the "." is used to access the instance variable (elemt) of an object, for example:

let obj = {
  cat: 'meow',
  dog: 'woof'
};
let sound = obj.cat;
console.log(sound);
// "meow is returned"

But in Solidity, the Dot Notation "." sometimes seems to be used to access elements in an object/struct, and sometimes it seems to be used for executing a function on a state variable, for example:

1. simpleStorageArray.push(simpleStorage); --> Using the "push" keyword to push simpleStorage contract into simpleStorageArray
2. SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).store(_simpleStorageNumber); --> Using the "store" function to store "_simpleStorageNumber" into the SimpleStorage simpleStorageArray of the index "_simpleStorageIndex"
3. return SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).retrieve(); --> Retrieve "favoriteNumber" from SimpleStorage in the simpleStorageArray of the index "_simpleStorageIndex".

Hope you can help with my 2 questions! Thank you so much! :D

**Comment by cromewar** [comment]
Hello @widmung89 the reason is tuples are not proper types in Solidity, they can only be used to form syntactic groupings of expressions. So as you can see the function `latestRoundData` returns a bunch of different information, and each one has a different type, uint80 is not the same as uint256. Solidity is a strongly typed language so each variable should be declared independently with it's type, that's why we do that. Also you will avoid later conflicts managing types on your contracts.

Regarding to the dot notation, you actually are just asking for functions and nothing more: `push` is a function with comes natively with the array object, `store` and `retrieve` are the functions we declare on the contract . But all of them are actually functions.

I hope this info might help. Best regards, Cromewar.
